lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -49,15 +49,20 @@ export class PublicClientApplication extends ClientApplication implements IPubli
         this.initializeBrokering();
     }
 
+    /**
+     * 
+     */
     private initializeBrokering(): void {
         if (this.config.system.brokerOptions.actAsBroker) {
-            this.broker = new BrokerManager(this.config.system.brokerOptions, this.logger, version);
-            this.logger.verbose("Acting as Broker");
-            this.broker.listenForMessage();
+            this.broker = new BrokerClientApplication(this.config);
         } else if (this.config.system.brokerOptions.allowBrokering) {
-            this.embeddedApp = new BrokerClient(this.config.system.brokerOptions, this.logger, this.config.auth.clientId,  version);
+            this.embeddedApp = new EmbeddedClientApplication(this.config.system.brokerOptions, this.logger, this.config.auth.clientId, version, this.browserStorage);
```

Comment:
Any reason we aren't just passing the whole config object here?

Reply:
Mostly because we aren't really using much from there aside from clientid and brokerOptions, but we can pass the entire config if it's easier.

Reply:
I think it may make it cleaner and more future-proof if we do need to access more config options in the future.

Reply:
done.